piptools/scripts/sync.py

```diff
@@ -125,7 +131,13 @@ def cli(
         log.error(str(e))
         sys.exit(2)
 
-    installed_dists = get_installed_distributions(skip=[], user_only=user_only)
+    paths = None
+    if python_executable:
+        paths = get_sys_path_for_python_executable(python_executable)
```

Comment:
I would rewrite this to:

```suggestion
    paths = None if python_executable is None else get_sys_path_for_python_executable(python_executable) 
```

or vice versa